main/src/ca/uwaterloo/flix/language/ast/ops/SimplifiedAstOps.scala

```diff
@@ -513,4 +515,291 @@ object SimplifiedAstOps {
     root
   }
 
+  def renameBoundVars(exp0: Expression, vars0: Map[Symbol.VarSym, Symbol.VarSym], labels0: Map[Symbol.LabelSym, Symbol.LabelSym])(implicit flix: Flix): Expression =
+    exp0 match {
+      case Expression.Unit => exp0
+      case Expression.True => exp0
+      case Expression.False => exp0
+      case Expression.Char(lit) => exp0
+      case Expression.Float32(lit) => exp0
+      case Expression.Float64(lit) => exp0
+      case Expression.Int8(lit) => exp0
+      case Expression.Int16(lit) => exp0
+      case Expression.Int32(lit) => exp0
+      case Expression.Int64(lit) => exp0
+      case Expression.BigInt(lit) => exp0
+      case Expression.Str(lit) => exp0
+
+      case Expression.Def(sym, tpe, loc) => Expression.Def(sym, tpe, loc)
+
+      case Expression.Var(sym, tpe, loc) =>
+        val s = vars0.getOrElse(sym, sym) // sym might not have been renamed e.g. if it's a free variable.
+        Expression.Var(s, tpe, loc)
+
+      case Expression.Lambda(args0, body, tpe, loc) =>
+        val args1 = args0.map(fparam => fparam.copy(sym = Symbol.freshVarSym(fparam.sym)))
```

Comment:
Why does this "refresh" a variable and not use vars0. It seems inconsistent. 

(Apologies if this code was not written by you, but we should get it sorted.)

Reply:
It was written by me :)

The idea is that renameBoundVars handles the renaming of all variables that are bound in exp0. And vars0 is only for looking up free variables. So when renaming the body of the lambda, the lambda-bound variables are free in that expression and must be looked up (hence we augment vars0). Conversely, when renaming the lambda expression, those variables are not free and hence must not be in vars0.

Reply:
OK; some documentation would make this clear.